### PR TITLE
 EZP-29812: As an editor, I want to embed a content item within a block element

### PR DESCRIPTION
**Pull request by dew326**
| Question      | Answer
| ------------- | ---
| Tickets       | https://jira.ez.no/browse/EZP-29812
| Requires       | https://github.com/ezsystems/ezplatform-richtext/pull/23
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review
- [x] frontend
- [x] backend

### CODE REVIEW

File: src/bundle/Resources/public/js/alloyeditor/src/toolbars/config/ez-embed-inline.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import EzConfigBase from './base';
+
+export default class EzEmbedInlineConfig extends EzConfigBase {
+    constructor() {
```
Comment (sunpietro) [inline src/bundle/Resources/public/js/alloyeditor/src/toolbars/config/ez-embed-inline.js]:
```
constructor(props) {
    super(props);
```

File: src/bundle/Resources/public/js/alloyeditor/src/toolbars/config/ez-embed-inline.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        const nativeEvent = payload.data.nativeEvent;
+
+        if (!nativeEvent) {
+            return false;
+        }
+
+        const target = new CKEDITOR.dom.element(nativeEvent.target);
+        const widget = payload.editor.get('nativeEditor').widgets.getByElement(target);
+
+        return !!(widget && widget.name === 'ezembedinline');
```
Comment (sunpietro) [inline src/bundle/Resources/public/js/alloyeditor/src/toolbars/config/ez-embed-inline.js]: `'ezembedinline'` could be stored as const variable.

File: src/bundle/Resources/public/js/alloyeditor/src/widgets/ez-embed-base.js [hunk 3]
```diff
@@ -0,0 +1,727 @@
...
+        const request = new Request('/api/ezp/v2/views', {
+            method: 'POST',
+            headers: {
+                Accept: 'application/vnd.ez.api.View+json; version=1.1',
+                'Content-Type': 'application/vnd.ez.api.ViewInput+json; version=1.1',
+                'X-Siteaccess': siteaccess,
+                'X-CSRF-Token': token,
+            },
+            body,
+            mode: 'cors',
```
Comment (sunpietro) [inline src/bundle/Resources/public/js/alloyeditor/src/widgets/ez-embed-base.js]: `same-origin` maybe?

File: src/bundle/Resources/public/js/alloyeditor/src/widgets/ez-embed-base.js [hunk 4]
```diff
@@ -0,0 +1,727 @@
...
+        const siteaccess = document.querySelector('meta[name="SiteAccess"]').content;
+        const request = new Request(variationHref, {
+            method: 'GET',
+            headers: {
+                Accept: 'application/vnd.ez.api.ContentImageVariation+json',
+                'X-Siteaccess': siteaccess,
+                'X-CSRF-Token': token,
+            },
+            credentials: 'same-origin',
+            mode: 'cors',
```
Comment (sunpietro) [inline src/bundle/Resources/public/js/alloyeditor/src/widgets/ez-embed-base.js]: `same-origin` maybe?

File: src/bundle/Resources/public/js/alloyeditor/src/widgets/ez-embed-base.js [hunk 5]
```diff
@@ -0,0 +1,727 @@
...
+        const siteaccess = document.querySelector('meta[name="SiteAccess"]').content;
+        const request = new Request(currentVersionHref, {
+            method: 'GET',
+            headers: {
+                Accept: 'application/vnd.ez.api.Version+json',
+                'X-Siteaccess': siteaccess,
+                'X-CSRF-Token': token,
+            },
+            credentials: 'same-origin',
+            mode: 'cors',
```
Comment (sunpietro) [inline src/bundle/Resources/public/js/alloyeditor/src/widgets/ez-embed-base.js]: `same-origin` maybe?

**Comment by dew326** [comment]
@katarzynazawada 1, 2, 3, 4, 5 - fixed (please update also branch in richtext bundle)
6 as discussed in private have to be done in a separate ticket (we have the same issue for the custom tags in tables)

**Comment by lserwatka** [comment]
@katarzynazawada please re-test is as soon as possible, as I want to merge it for beta later today.

**Pull request merged**